rclpy/rclpy/executors.py

```diff
@@ -700,7 +720,10 @@ class SingleThreadedExecutor(Executor):
     def __init__(self, *, context: Context = None) -> None:
         super().__init__(context=context)
 
-    def spin_once(self, timeout_sec: float = None) -> None:
+    def _spin_once_impl(
+        self,
+        timeout_sec: Optional[Union[float, TimeoutObject]] = None
+    ) -> None:
```

Comment:
So this private method is new for Iron? since this is private i think that should be no problem? or for iron and humble, we should add `Option` to `spin_once` directly?

Reply:
@iuhilnehc-ynos can you also review this?

Reply:
> So this private method is new for Iron? since this is private i think that should be no problem? or for iron and humble, we should add Option to spin_once directly?

For rolling, we keep spin_once only accept float parameter as timeout. So I don't want to change spin_once interface to support timeout object. This can keep the same interface as rolling.

Reply:
LGTM.

BUT I think https://github.com/ros2/rclpy/pull/1143 should be backported before https://github.com/ros2/rclpy/pull/1165, otherwise the the backported for https://github.com/ros2/rclpy/pull/1143 needs to fix conflict again.